### PR TITLE
Implement Verify API

### LINKED ISSUE #37: Implement Verify API
Currently it seems like the verify API is missing: https://www.twilio.com/verify

It would be very helpful to have it implemented in this package as it's a very common feature for registration

### PR DESCRIPTION
**Pull request by eloyekunle**
Fixes #37

### CODE REVIEW

File: verify.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+type VerifyPhoneNumberService struct {
+	client *Client
+}
+
+type VerifyPhoneNumber struct {
+	Sid         string      `json:"sid"`
+	ServiceSid  string      `json:"service_sid"`
+	AccountSid  string      `json:"account_sid"`
+	To          string      `json:"to"`
```
Comment (kevinburke) [inline verify.go]: Should this be a PhoneNumber type?

File: verify.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+type CheckPhoneNumber struct {
+	Sid         string     `json:"sid"`
+	ServiceSid  string     `json:"service_sid"`
+	AccountSid  string     `json:"account_sid"`
+	To          string     `json:"to"`
+	Channel     string     `json:"channel"`
+	Status      string     `json:"status"`
+	Valid       bool       `json:"valid"`
+	Amount      string     `json:"amount"`
```
Comment (kevinburke) [inline verify.go]: types.NullString here, I think, based on the response.

File: verify.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	Payee       string     `json:"payee"`
```
Comment (kevinburke) [inline verify.go]: Same here.

File: verify.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	DateCreated TwilioTime `json:"date_created"`
+	DateUpdated TwilioTime `json:"date_updated"`
+}
+
+func (v *VerifyPhoneNumberService) Create(ctx context.Context, verifyServiceID string, data url.Values) (*VerifyPhoneNumber, error) {
```
Comment (kevinburke) [inline verify.go:50]: Can you add a docstring that links to the Twilio docs?

File: verify_test.go [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+	t.Parallel()
+	client, s := getServer(verifyResponse)
+	defer s.Close()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	data := url.Values{}
+	data.Add("to", "+14159373912")
+	data.Add("channel", "sms")
+	verify, err := client.Verify.VerifyPhoneNumbers.Create(ctx, "VA9e0bd45bfa7d9b9e7dca86cf94c7d4f8", data)
```
Comment (kevinburke) [inline verify_test.go]: Not a huge fan of the "Verify.Verify" stuttering here... maybe can we at least do "client.Verify.Verifications.Create" ?

Other thoughts on the API?

File: verify_test.go [hunk 6]
```diff
@@ -0,0 +1,88 @@
+package twilio
+
+import (
+	"context"
+	"net/url"
+	"testing"
+)
+
+func TestVerifyPhoneNumbersCreate(t *testing.T) {
```
Comment (kevinburke) [inline verify_test.go:9]: Thank you for adding tests!

File: verify.go (hunk 1 above)
Comment (eloyekunle) [reply verify.go]: Yes

File: verify_test.go (hunk 6 above)
Comment (eloyekunle) [reply verify_test.go:9]: You're welcome.

File: verify.go (hunk 1 above)
Comment (eloyekunle) [reply verify.go]: Fixed.

File: verify.go (hunk 2 above)
Comment (eloyekunle) [reply verify.go]: Fixed.

File: verify.go (hunk 3 above)
Comment (eloyekunle) [reply verify.go]: Fixed

File: verify.go (hunk 4 above)
Comment (eloyekunle) [reply verify.go:50]: Done

File: verify_test.go (hunk 5 above)
Comment (eloyekunle) [reply verify_test.go]: I've renamed it.

The API looks good to me.

**Review by kevinburke — APPROVED** [review]
LGTM

**Comment by kevinburke** [comment]
Merged as a697940198d87b517a95187b03e6017094963247! Thanks!

**Pull request closed without merging**